File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    let answer = 0;
    let item_dict = {};
    for (let i = 0; i < number.length; i++) item_dict[want[i]] = number[i];
    
    for (let i = 0; i <= discount.length-10; i++) {
        let slice = discount.slice(i, i+10);
        let flag = true;
        
        for (let j of want) {
            if (slice.filter((v) => v === j).length !== item_dict[j]) {
                flag = false;
                break;
            }
        }
        if (flag) answer++;
    }
    
  return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter

def solution(want, number, discount):
    answer = 0
    item_dict = {v: n for v,n in zip(want, number)}

    for i in range(len(discount)-9):
        if item_dict == Counter(discount[i:i+10]): answer += 1

    return answer